js/DropZone/DropZoneComponent.js

```diff
@@ -0,0 +1,564 @@
+            interactionClasses: {
+                windowEnter: 'dropzone-window-active',
+                dropZoneEnter: 'dropzone-dropzone-active',
+                dropZoneSuccess: 'dropzone-success',
+                dropZoneError: 'dropzone-error'
+            }
+        };
+
+        const callbacks = {
+            windowEnter: this.handleWindowEnter.bind(this),
+            windowLeave: this.handleWindowLeave.bind(this),
+            dropZoneEnter: this.handleDropZoneEnter.bind(this),
+            dropZoneLeave: this.handleDropzoneLeave.bind(this),
+            dropZoneDrop: this.handleDropZoneDrop.bind(this),
+            windowDrop: this.handleWindowDrop.bind(this),
+            fileRemoved: this.handleFileRemoved.bind(this)
+        };
+
+        // get all DropZone elements
+        this.dropZoneInstances = [].slice.call(this.html.querySelectorAll(this.selector));
+
+        // build options from options passed to the constructor and the defaults
+        this.optionsManager.buildComponentOptions(options, defaults, callbacks);
+
+        // add DropZone(s) to instance manager
+        this.dropZoneInstances.forEach(node => {
+            // create and store DropZone instance
+            this.instanceManager.addInstance(node, this.optionsManager);
+        });
+
+        // iterate over our DropZoneInstanceManagerInstances to process any input nodes & update help state
+        this.instanceManager.getInstance().forEach(instance => {
+            if (instance.input) {
+                this.processInputNode(instance.input, instance.id, instance.options.showInputNode);
+
+                if (instance.browse) {
+                    this.processBrowseNode(instance.browse, instance.input);
+                }
+            }
+        });
+
+        // bind this here, so we have a ref
+        this.removeFile = this.removeFile.bind(this);
+    }
+
+    /**
+     * If an input node ID has been passed in, we'll take care of linking the DropZone
+     * instance to this node
+     * @param {Element} input
+     * @param {number} id
+     * @param {boolean} show
+     */
+    processInputNode (input, id, show) {
+        // for now these handlers can stay anonymous, the likely hood of wanting to disable a DropZone
+        // at some point in a session seems unlikely. famous. last. words.
+        input.addEventListener('change', () => {
+            const files = input.files;
+
+            // we'll add a persist property to our file objects, this can be used to
+            // persist the front-end validation, which is essential when using an
+            // associated file input
+            if (input.value) {
+                this.instanceManager.addFiles(files, id, { persist: true });
+            }
+
+            // reset input node value, this will ensure our change event
+            // fires each time we use the browse files functionality - even
+            // if we try to add an identical value
+            input.value = '';
```

Comment:
It's only IE9 that kicks off about that (I believe), in which case we don't need to reset the input as it doesn't support any enriched features. I could make that clearer by putting this in the `if (this.supported)` condition.